ton_client/src/tvm/call_tvm.rs

```diff
@@ -16,35 +16,37 @@ use super::types::ResolvedExecutionOptions;
 use crate::error::ClientResult;
 use crate::tvm::Error;
 use std::sync::Arc;
-use ton_block;
-use ton_block::{Deserializable, Message, Serializable};
+use ton_block::{
+    Account, ConfigParams, CurrencyCollection,
+    CommonMsgInfo, Deserializable, Serializable,
+    OutActions, OutAction,
+    Message, MsgAddressInt,
+};
 use ton_types::dictionary::HashmapType;
 use ton_types::SliceData;
 use ton_vm::executor::gas::gas_state::Gas;
 use ton_vm::stack::{integer::IntegerData, savelist::SaveList, Stack, StackItem};
 
 pub(crate) fn call_tvm(
-    mut account: ton_block::AccountStuff,
+    account: &mut Account,
     options: ResolvedExecutionOptions,
     stack: Stack,
-) -> ClientResult<(ton_vm::executor::Engine, ton_block::AccountStuff)> {
-    let mut state = match &mut account.storage.state {
-        ton_block::AccountState::AccountActive(state) => Ok(state),
-        _ => Err(Error::invalid_account_boc("Account is not active")),
-    }?;
+) -> ClientResult<ton_vm::executor::Engine> {
+    let code = account.get_code().unwrap_or_default();
+    let data = account.get_data().ok_or_else(|| Error::invalid_account_boc("Account has no code"))?;
 
     let mut ctrls = SaveList::new();
     ctrls
         .put(
             4,
-            &mut StackItem::Cell(state.data.clone().unwrap_or_default()),
+            &mut StackItem::Cell(data),
         )
         .map_err(|err| Error::internal_error(format!("can not put data to registers: {}", err)))?;
 
     let sci = build_contract_info(
         options.blockchain_config.raw_config(),
-        &account.addr,
-        &account.storage.balance,
+        account.get_addr().unwrap(),
+        account.balance().unwrap(),
```

Comment:
Can we be sure that these values are always set?

Reply:
unwrap removed

Reply:
removed